homeassistant/components/axis/light.py

```diff
@@ -1,75 +1,114 @@
+    light_type = hub.api.vapix.light_control[light_id].light_type
+    return f"{light_type} {event_type} {event.id}"
+
+
+@dataclass(frozen=True, kw_only=True)
+class AxisLightDescription(LightEntityDescription):
+    """Axis light entity description."""
+
+    event_topic: EventTopic
+    """Event topic that provides state updates."""
+    name_fn: Callable[[AxisHub, Event], str]
+    """Function providing the corresponding name to the event ID."""
+    supported_fn: Callable[[AxisHub, Event], bool]
+    """Function validating if event is supported."""
+
+
+ENTITY_DESCRIPTIONS = (
+    AxisLightDescription(
+        key="Light state control",
+        event_topic=EventTopic.LIGHT_STATUS,
+        name_fn=light_name_fn,
+        supported_fn=lambda hub, event: len(hub.api.vapix.light_control) > 0,
+    ),
+)
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up a Axis light."""
+    """Set up the Axis light platform."""
     hub = AxisHub.get_hub(hass, config_entry)
 
-    if hub.api.vapix.light_control is None or len(hub.api.vapix.light_control) == 0:
-        return
-
     @callback
-    def async_create_entity(event: Event) -> None:
-        """Create Axis light entity."""
-        async_add_entities([AxisLight(event, hub)])
+    def register_platform(descriptions: Iterable[AxisLightDescription]) -> None:
+        """Register entity platform to create entities on event initialized signal."""
+
+        @callback
+        def create_entity(description: AxisLightDescription, event: Event) -> None:
+            """Create Axis entity."""
+            if description.supported_fn(hub, event):
+                async_add_entities([AxisLight(hub, description, event)])
+
+        for description in descriptions:
+            hub.api.event.subscribe(
+                partial(create_entity, description),
+                topic_filter=description.event_topic,
+                operation_filter=EventOperation.INITIALIZED,
+            )
 
-    hub.api.event.subscribe(
-        async_create_entity,
-        topic_filter=EventTopic.LIGHT_STATUS,
-        operation_filter=EventOperation.INITIALIZED,
-    )
+    register_platform(ENTITY_DESCRIPTIONS)
 
 
 class AxisLight(AxisEventEntity, LightEntity):
-    """Representation of a light Axis event."""
+    """Representation of a Axis light."""
 
     _attr_should_poll = True
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
 
-    def __init__(self, event: Event, hub: AxisHub) -> None:
+    def __init__(
+        self, hub: AxisHub, description: AxisLightDescription, event: Event
+    ) -> None:
         """Initialize the Axis light."""
         super().__init__(event, hub)
+        self.entity_description = description
+        self._attr_name = description.name_fn(hub, event)
+        self._attr_is_on = event.is_tripped
```

Comment:
Would it make sense to extract the setting of the `_attr` in a separate function which you can also use at L116? In this case we're only talking about 1 shared line, but it would avoid more DRY (or bugs) when in the future you want to add more attributes (so you always have 1 shared place where you set the state of the entity)

Reply:
Absolutely. There are probably more stuff that can be simplified over time. Right my main focus is to get a common baseline with entity descriptions. But this should definitely be addressed in the future